pkg/db/db.go

```diff
@@ -52,6 +53,11 @@ func (db *RDB) ExecContext(ctx context.Context, query string, args ...interface{
 	}
 }
 
+// QueryRowContext wraps sql.DB.QueryRowContext.
+func (db *RDB) QueryRowContext(ctx context.Context, query string, args ...interface{}) *sql.Row {
+	return db.DB.QueryRowContext(ctx, query, args...)
```

Comment:
Errors are deferred until scan so we can't check for transaction timeouts here

Reply:
I think either;

* Use `QueryContext` and assume one row and make it retry on lock timeouts.
* Create a wrapper interface for `*sql.Row` like we did for `*sql.DB` and then have a `Scan` method that retries on lock timeouts.* Add a `GetDB()` method to the `DB` interface to return the underlying `*sql.DB` and call `QueryRowContext` on that - that avoids setting the false expectation that the query will be retried on lock timeouts.

Reply:
I'll go with the first option as it the most consistent with what we currently have.